libraries/griptape_nodes_library/griptape_nodes_library/engine/list_files.py

```
@@ -83,10 +109,54 @@ def __init__(
+        for entry in entries:
+            if entry.is_dir:
+                if include_folders:
+                    filtered_entries.append(entry)
+            elif include_files:
+                filtered_entries.append(entry)
+        return filtered_entries
+
+    def _convert_paths(self, entries: list, *, use_absolute_paths: bool) -> list[str]:
+        """Extract paths from entries, optionally converting to absolute paths."""
+        file_paths = []
+        for entry in entries:
+            if use_absolute_paths:
+                entry_path = Path(entry.path)
+                if not entry_path.is_absolute():
+                    workspace_path = GriptapeNodes.ConfigManager().workspace_path
+                    absolute_path = (workspace_path / entry_path).resolve()
+                    file_paths.append(str(absolute_path))
+                else:
+                    file_paths.append(entry.path)
+            else:
+                file_paths.append(entry.path)
+        return file_paths
+
     def process(self) -> None:
         self._clear_execution_status()
         directory_path = self.get_parameter_value("directory_path")
         show_hidden = self.get_parameter_value("show_hidden")
+        list_options = self.get_parameter_value("list_options")
+        use_absolute_paths = self.get_parameter_value("use_absolute_paths")
+
+        # Determine include_files and include_folders based on list_options
+        if list_options == LIST_OPTIONS[0]:  # "List files and folders"
+            include_files = True
+            include_folders = True
+        elif list_options == LIST_OPTIONS[1]:  # "List files only"
+            include_files = True
+            include_folders = False
+        elif list_options == LIST_OPTIONS[2]:  # "List folders only"
+            include_files = False
+            include_folders = True
+        else:
+            # Fallback to default if invalid option
+            include_files = True
+            include_folders = True
```

Since this parameter is an input, we should probably throw here in case the user gave an invalid value. match/case ftw.